### PR TITLE
HAMR-179 Add pyproject.toml to fix docker build

### PR DESCRIPTION
**Pull request by michael-richey**
### What does this PR do?
Fixes the `docker build` command which is currently broken.

### Description of the Change
Creates a skeleton `pyproject.toml` file so the build will work.

### Alternate Designs
We could use a different build backend. For example, passing `--use-pep517` would fix it but swapping the build backend seems like a larger change. Changing the base image (to python3.12-slim-bullseye) also seems to resolve but again it's a larger change. 

### Verification Process
Tested the documented `docker build` command then ran an `import` and `sync` with the newly created docker image.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## 0.11.1 / 2024-09-04
+
+### Fixed
+* fixed the `docker build` command
```
Comment (drichards-87) [inline CHANGELOG.md:6]:
```suggestion
* Fix the `docker build` command
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+### Changed
+* added the `pyproject.toml` file by @michael-richey in https://github.com/DataDog/datadog-sync-cli/pull/271
```
Comment (drichards-87) [inline CHANGELOG.md:8]:
```suggestion
* Add the `pyproject.toml` file by @michael-richey in https://github.com/DataDog/datadog-sync-cli/pull/271
```

**Review by drichards-87 — APPROVED** [review]
Left a couple of small suggestions from Docs and approved the PR.

File: CHANGELOG.md (hunk 1 above)
Comment (michael-richey) [reply CHANGELOG.md]: Changed to match yours

File: CHANGELOG.md (hunk 2 above)
Comment (michael-richey) [reply CHANGELOG.md]: Changed to match yours

**Review by heyronhay — DISMISSED** [review]

**Review by skarimo — APPROVED** [review]

**Pull request merged**